flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/batch/sql/TimeTravelITCase.java

```diff
@@ -0,0 +1,305 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.runtime.batch.sql;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.planner.factories.TestTimeTravelCatalog;
+import org.apache.flink.table.planner.factories.TestValuesTableFactory;
+import org.apache.flink.table.planner.runtime.utils.BatchTestBase;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CollectionUtil;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/** Test for time travel. */
+public class TimeTravelITCase extends BatchTestBase {
+    private static final List<Tuple3<String, Schema, List<Row>>> TEST_TIME_TRAVEL_DATE =
+            Arrays.asList(
+                    Tuple3.of(
+                            "2023-01-01 01:00:00",
+                            Schema.newBuilder().column("f1", DataTypes.INT()).build(),
+                            Collections.singletonList(Row.of(1))),
+                    Tuple3.of(
+                            "2023-01-01 02:00:00",
+                            Schema.newBuilder()
+                                    .column("f1", DataTypes.INT())
+                                    .column("f2", DataTypes.INT())
+                                    .build(),
+                            Collections.singletonList(Row.of(1, 2))),
+                    Tuple3.of(
+                            "2023-01-01 03:00:00",
+                            Schema.newBuilder()
+                                    .column("f1", DataTypes.INT())
+                                    .column("f2", DataTypes.INT())
+                                    .column("f3", DataTypes.INT())
+                                    .build(),
+                            Collections.singletonList(Row.of(1, 2, 3))));
+
+    private static final List<Tuple2<String, String>> EXPECTED_TIME_TRAVEL_RESULT =
+            Arrays.asList(
+                    Tuple2.of("2023-01-01 01:00:00", "[+I[1]]"),
+                    Tuple2.of("2023-01-01 02:00:00", "[+I[1, 2]]"),
+                    Tuple2.of("2023-01-01 03:00:00", "[+I[1, 2, 3]]"));
+
+    @BeforeEach
+    public void before() {
```

Comment:
nit:
```suggestion
    void before() {
```

Reply:
Due to the limitations of the parent class, we can't remove "public" here.